### ISSUE TITLE
Using TypeChecker to read JSDoc @param type from function parameter

### ISSUE THREAD

**Issue by Madd0g** [Microsoft/TypeScript#23395]
**TypeScript Version:**  2.9.0-dev.20180330

**Search Terms:** typechecker getTypeFromTypeNode jsdoc object

**Code**

```ts
/**
 * my description
 * @param {object} opts options
 */
function myFunc(opts) {
    //
}

```

I create a `Program`, then with its `TypeChecker`, I try to read the type of the param:

```ts
let checker = program.getTypeChecker();
let p = funcNode.parameters[0];
let paramJSDocTypeNode = ts.getJSDocType(p);
let type = checker.getTypeFromTypeNode(paramJSDocTypeNode);
```

**Expected behavior:** No error.

**Actual behavior:**
```ts
TypeError: Cannot read property 'flags' of undefined
    at getMembersOfSymbol (/mytest/node_modules/typescript/lib/typescript.js:30759:26)
    at getTypeFromTypeLiteralOrFunctionOrConstructorTypeNode (/mytest/node_modules/typescript/lib/typescript.js:33419:21)
    at getTypeFromTypeNode (/mytest/node_modules/typescript/lib/typescript.js:33672:28)
    at Object.getTypeFromTypeNode (/mytest/node_modules/typescript/lib/typescript.js:25738:31)
    at params.map.p (/mytest/read-ts.ts:1:25774)
```

If I change `{object}` to `{string}` it doesn't throw that error.

**Comment by mhegazy** [comment]
how did you create the program? do you have `allowjs` enabled, and does the filename has `.js` in it?

**Comment by Madd0g** [comment]
@mhegazy - I mostly followed this example in "[Using the Compiler API](https://github.com/Microsoft/TypeScript-wiki/blob/master/Using-the-Compiler-API.md#incremental-build-support-using-the-language-services)" - I create a `LanguageServiceHost` and then a `LanguageService` and then I get the checker like this `services.getProgram().getTypeChecker()`.

I do have `allowJs` and `checkJs`. The specific file I worked on was a `.ts` file.

**Comment by mhegazy** [comment]
Types are not read in a .ts file from jsdoc nodes. so that code path is not supported.

**Comment by mhegazy** [comment]
the rational is you already have a syntax for writing types, the jsdoc is rather redundant. 

the comments are there though, and you can convert the types to TS types in some cases, see https://github.com/Microsoft/TypeScript/blob/master/src/services/codefixes/fixJSDocTypes.ts for an example.

**Comment by Madd0g** [comment]
I understand why the decision was made, I think it would still be cool if by default TS could point out errors in JSDoc anyway. I think of TS as a step in a pipeline and since it already knows how to parse JSDocs, it's a hassle to involve another tool or custom code for this, later in the pipeline, when you can get it right there in the editor from TS.
 
Thank you for the example, I don't fully understand it yet, but it is helpful.
